src/util/sockopt.rs

```diff
@@ -101,24 +101,46 @@ mod tests {
     fn test_sock_opt() {
         let addr = "127.0.0.1:0".parse().unwrap();
 
-        let server = TcpListener::bind(&addr).unwrap();
-        let addr = &format!("{}", server.local_addr().unwrap());
-
-        let sock = TcpStream::connect(&addr.parse().unwrap()).unwrap();
+        // mio
```

Comment:
Why not remove the test attribute and create functions `test_mio_sock_opt` and `test_std_sock_opt` call this function instead?